tests/integration/test_index_routines.py

```diff
@@ -36,6 +36,11 @@ def test_choose_1d():
     num_a1 = num.array(a1)
     num_choices1 = num.array(choices1)
 
+    assert np.array_equal(
+        num.choose(num_a1, num_choices1, out=None),
+        np.choose(a1, choices1, out=None),
+    )
+
```

Comment:
As long as we are already touching this test, it really needs to be split into five tests:

* 2 for with `out` vs without `out`
* 3 for mode: default, "clip", and "wrap"

Reply:
Fixed